src/components/common/EntityActiveToggle.test.tsx

```diff
@@ -0,0 +1,251 @@
+import React from "react"
+import { render, screen, fireEvent, waitFor } from "@testing-library/react"
+import { EntityActiveToggle } from "./EntityActiveToggle"
+
+const mockHandleChangeAndSave = jest.fn()
+const mockToastSuccess = jest.fn()
+const mockToastError = jest.fn()
+
+// Mock the contexts
+jest.mock("@/contexts", () => ({
+  useClient: () => ({
+    user: mockUser,
+    client: {},
+  }),
+  useCampaign: () => ({
+    campaign: mockCampaign,
+  }),
+  useToast: () => ({
+    toastSuccess: mockToastSuccess,
+    toastError: mockToastError,
+  }),
+}))
+
+let mockUser = { id: "1", admin: false, gamemaster: false }
+let mockCampaign = { id: "campaign1", gamemaster_id: "2" }
+
+describe("EntityActiveToggle", () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockUser = { id: "1", admin: false, gamemaster: false }
+    mockCampaign = { id: "campaign1", gamemaster_id: "2" }
+  })
+
+  describe("Permission visibility", () => {
+    it("should not render for regular users", () => {
+      render(
+        <EntityActiveToggle
+          entity={{ id: "123", entity_class: "Character", active: true }}
+          handleChangeAndSave={mockHandleChangeAndSave}
+        />
+      )
+
+      expect(screen.queryByRole("checkbox")).not.toBeInTheDocument()
+    })
```

Comment:
This is brittle, it would fail if we put some unrelated checkbox on the document.